x-pack/plugins/data_visualizer/public/application/common/components/document_count_content/document_count_content.tsx

```diff
@@ -5,17 +5,93 @@
+import { sortedIndex } from 'lodash';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { isDefined } from '../../util/is_defined';
+import type { DocumentCountChartPoint } from './document_count_chart';
+import {
+  RANDOM_SAMPLER_STEP,
+  RANDOM_SAMPLER_PROBABILITIES,
+  RandomSamplerOption,
+  RANDOM_SAMPLER_SELECT_OPTIONS,
+  RANDOM_SAMPLER_OPTION,
+} from '../../../index_data_visualizer/constants/random_sampler';
 import { TotalCountHeader } from './total_count_header';
-import { DocumentCountStats } from '../../../../../common/types/field_stats';
+import type { DocumentCountStats } from '../../../../../common/types/field_stats';
+import { DocumentCountChart } from './document_count_chart';
 
 export interface Props {
   documentCountStats?: DocumentCountStats;
   totalCount: number;
+  samplingProbability?: number | null;
+  setSamplingProbability?: (value: number) => void;
+  randomSamplerPreference?: RandomSamplerOption;
+  setRandomSamplerPreference: (value: RandomSamplerOption) => void;
+  loading: boolean;
 }
 
-export const DocumentCountContent: FC<Props> = ({ documentCountStats, totalCount }) => {
+export const DocumentCountContent: FC<Props> = ({
+  documentCountStats,
+  totalCount,
+  samplingProbability,
+  setSamplingProbability,
+  loading,
+  randomSamplerPreference,
+  setRandomSamplerPreference,
+}) => {
+  const [showSamplingOptionsPopover, setShowSamplingOptionsPopover] = useState(false);
+
+  const onShowSamplingOptions = useCallback(() => {
+    setShowSamplingOptionsPopover(!showSamplingOptionsPopover);
+  }, [showSamplingOptionsPopover]);
+
+  const closeSamplingOptions = useCallback(() => {
+    setShowSamplingOptionsPopover(false);
+  }, [setShowSamplingOptionsPopover]);
+
+  const calloutInfoMessage = useMemo(() => {
+    switch (randomSamplerPreference) {
+      case RANDOM_SAMPLER_OPTION.OFF:
+        return i18n.translate(
+          'xpack.dataVisualizer.randomSamplerSettingsPopUp.onManualCalloutMessage',
+          {
+            defaultMessage:
+              'Random sampling can be turned on for the total document count and chart to increase speed although some accuracy will be lost.',
+          }
+        );
+      case RANDOM_SAMPLER_OPTION.ON_AUTOMATIC:
+        return i18n.translate(
+          'xpack.dataVisualizer.randomSamplerSettingsPopUp.onAutomaticCalloutMessage',
+          {
+            defaultMessage:
+              'Random sampling is being used for the total document count and the chart. The probability used in the aggregation will be automatically set to balance accuracy and speed.',
+          }
+        );
+
+      case RANDOM_SAMPLER_OPTION.ON_MANUAL:
+      default:
+        return i18n.translate('xpack.dataVisualizer.randomSamplerSettingsPopUp.offCalloutMessage', {
+          defaultMessage:
+            'Random sampling is being used for the total document count and the chart. A lower percentage probability will increase performance, but some accuracy will be lost.',
```

Comment:
To match the other suggestion:
```suggestion
            'The total document count and chart use random sampler aggregations. A lower percentage probability increases performance, but some accuracy is lost.',
```

Reply:
Updated here https://github.com/elastic/kibana/pull/136150/commits/5959f472a0344c980918307d74fe4902005a40ad